### ISSUE TITLE
Support for yarn projects

### ISSUE THREAD

**Issue by lachrist** [getappmap/appmap-agent-js#168]
Yarn is an important package manager. And it is the one that we use for `appmap-js`. We should support yarn projects.

**Comment by dividedmind** [comment]
Also, pnpm.

**Comment by dustinbyrne** [comment]
What about `yarn` or other package managers is unsupported? Do we have specific tie ins with `npm` or `npx`?

**Comment by dividedmind** [comment]
@lachrist correct me if I'm wrong, but I believe the problem is that the agent parses the user-provided command line to figure out how to call and hook the project, and these heuristics only support npm at the moment; cf. https://github.com/getappmap/appmap-agent-js/blob/4ea08b38a953d7f89c2fac7a70adf27168931603/components/configuration-accessor/default/package.mjs#L48-L57

**Comment by lachrist** [comment]
Indeed. I haven't thought about supporting anything other than `npm` so I'm not sure what else is needed to support `yarn`. I suspect it is not much.
